applications/GeoMechanicsApplication/custom_utilities/stress_strain_utilities.cpp

```diff
@@ -52,26 +53,75 @@ double StressStrainUtilities::CalculateTrace(const Vector& StressVector)
     return trace;
 }
 
-double StressStrainUtilities::CalculateMeanStress(const Vector& StressVector)
+double StressStrainUtilities::CalculateMeanStress(const Vector& rStressVector)
 {
-    return CalculateTrace(StressVector) / (StressVector.size() == 3 ? 2.0 : 3.0);
+    return CalculateTrace(rStressVector) / (rStressVector.size() == 3 ? 2.0 : 3.0);
 }
 
-double StressStrainUtilities::CalculateVonMisesStrain(const Vector& StrainVector)
+double StressStrainUtilities::CalculateLodeAngle(const Vector& rStressVector)
 {
-    return (2.0 / 3.0) * CalculateVonMisesStress(StrainVector);
+    KRATOS_ERROR_IF(rStressVector.size() < 4);
+
+    const double p                   = CalculateMeanStress(rStressVector);
+    const double q                   = CalculateVonMisesStress(rStressVector);
+    Matrix       local_stress_tensor = MathUtils<double>::StressVectorToTensor(rStressVector);
```

Comment:
Maybe this one can be const too, depends a bit on the `GaussSeidelEigenSystem` function signature, but I'd expect this is the case.
```suggestion
    const double p                           = CalculateMeanStress(rStressVector);
    const double q                           = CalculateVonMisesStress(rStressVector);
    const Matrix local_stress_tensor = MathUtils<double>::StressVectorToTensor(rStressVector);
```

Reply:
Done